real_preprocessing/src/camera_pose.cpp

```diff
@@ -0,0 +1,263 @@
+  {
+    YAML::Node tags_pix = YAML::LoadFile("detections_" + std::to_string(filenum) + ".yaml");
+    Eigen::MatrixXd w_T_cam(4, 4);
+    if(world_pres==0)
+    {
+      std::vector<int>::iterator it = std::find(w_T_tags_id.begin(), w_T_tags_id.end(), tags_pix["detections"][loc]["targetID"].as<int>());
+      int index = std::distance(w_T_tags_id.begin(), it);
+      w_T_cam = w_T_tags_trans[index] * objTcam(loc, filenum);
+    }
+    if(world_pres==1)
+    {
+      w_T_cam=objTcam(loc,filenum);
+    }
+    worldAppend(filenum,w_T_cam);
+    for (int i = 0; i < tags_pix["detections"].size(); ++i)
+    {
+      if ((i != loc) &&
+          ((std::find(w_T_tags_id.begin(), w_T_tags_id.end(), tags_pix["detections"][i]["targetID"].as<int>()) !=
+            w_T_tags_id.end()) == false))
+      {
+        Eigen::MatrixXd obj_T_cam(4, 4);
+        obj_T_cam = objTcam(i, filenum);
+        w_T_tags_trans.push_back((w_T_cam * obj_T_cam.inverse().eval()));
+        w_T_tags_id.push_back(tags_pix["detections"][i]["targetID"].as<int>());
+        w_T_tags_size.push_back(tags_pix["detections"][i]["size"][0].as<double>());
+      }
+    }
+  }
+
+  //Returns status of file : world tag present, known tag present, all unknown tags, no file
+  int fileReader(int filenum)
+  {
+    bool known_tag(0);
+    std::ifstream fin;
+    fin.open("detections_" + std::to_string(filenum) + ".yaml");
+    if (fin.is_open())
+    {
+      fin.close();
+      YAML::Node tags_pix = YAML::LoadFile("detections_" + std::to_string(filenum) + ".yaml");
+      if (filenum == 0)//polling first file?
+      {
+        world = tags_pix["detections"][0]["targetID"].as<int>(); //world tag chosen in first file/first tag listing
+        world_loc = 0;
+        return 0; //world tag present
+      }
+      for (int i = 0; i < tags_pix["detections"].size(); ++i)
+      {
+        if (std::find(w_T_tags_id.begin(), w_T_tags_id.end(), tags_pix["detections"][i]["targetID"].as<int>()) !=w_T_tags_id.end())
+        {
+          known_tag = 1;
+          known_loc = i;
+        }
+        if (tags_pix["detections"][i]["targetID"].as<int>() == world) 
+        {
+          world_loc = i;
+          return 0; //world tag present
+        }
+      }
+      if (known_tag == 1)
+      {
+        return 1; //known tag present
+      }
+      else
+        return 2; //all unknown tags
+    }
+    else
+      return 3; //file does not exist - end of YAML list
+  }
+
+  //Polls for unknown files to be reprocessed 
+  void unknownPoll()
+  {
+    if (unknown_file.size() != 0) 
+      {
+        for (int i = (unknown_file.size() - 1); i >= 0; i--)
+        {
+          if (fileReader(unknown_file[i]) == 1)
+          {
+            tagCalc(unknown_file[i],known_loc,0);
+            unknown_file.erase(unknown_file.begin() + i);
+          }
+        }
+      }
+  }  
+
+  //streams all snapshot detection files for tag/pose processing 
+  void fileStream()
+  {
+    while (fileReader(file) != 3)
```

Comment:
The same file is being used in the same function over and over. Just store the fileReader(file) in a variable and reuse this.